doc/make_tables.py

```diff
@@ -0,0 +1,304 @@
+
+    @classmethod
+    def get_header(cls, data):
+        return cls.header
+
+    @classmethod
+    def get_row(cls, i, row_data):
+        if row_data["descr"] is None:
+            return None  # Skip color scheme if description is set to ``None``.
+        else:
+            # Create an image from the given color scheme and generate the row rst.
+            img_path = f"{CHARTS_IMAGE_DIR}/cs_{i}.png"
+            n_colors = cls.generate_img(row_data["scheme"], img_path)
+            return cls.row_template.format(
+                row_data["scheme"], row_data["descr"], n_colors, img_path
+            )
+
+    @staticmethod
+    def generate_img(color_scheme, img_path):
+        """Generate and save an image of the given color_scheme."""
+        p = pv.Plotter(off_screen=True, window_size=[240, 120])
+        p.background_color = 'w'
+        chart = pv.Chart2D()
+        # Use a temporary plot to determine the total number of colors in this scheme
+        tmp_plot = chart.bar([0], [[1]] * 2, color=color_scheme, orientation="H")
+        n_colors = len(tmp_plot.colors)
+        plot = chart.bar([0], [[1]] * n_colors, color=color_scheme, orientation="H")
+        chart.remove_plot(tmp_plot)
+        plot.pen.color = 'w'
+        chart.x_range = [0, n_colors]
+        chart.hide_axes()
+        p.add_chart(chart)
+        _, img = p.show(screenshot=True, return_cpos=True)
+        # Crop the image and save it
+        p._save_image(img[34:78, 22:225, :], img_path, False)
+        return n_colors
+
+
+class ColorTable(DocTable):
+    """Class to generate colors table."""
+
+    path = f"{COLORS_TABLE_DIR}/colors.rst"
+    header = _aligned_dedent(
+        """
+        |.. list-table::
+        |   :widths: 50 20 30
+        |   :header-rows: 1
+        |
+        |   * - Name
+        |     - Hex value
+        |     - Example
+        """
+    )
+    row_template = _aligned_dedent(
+        """
+        |   * - {}
+        |     - ``{}``
+        |     - .. raw:: html
+        |
+        |          <span style='width:100%; height:100%; display:block; background-color: {};'>&nbsp;</span>
+        """
+    )
+
+    @classmethod
+    def fetch_data(cls):
+        # Fetch table data from ``hexcolors`` dictionary.
+        colors = {
+            name: {"name": name, "hex": hex, "synonyms": []} for name, hex in pv.hexcolors.items()
+        }
+        # Add synonyms defined in ``color_synonyms`` dictionary.
+        for s, name in pv.colors.color_synonyms.items():
+            colors[name]["synonyms"].append(s)
+        return colors.values()
+
+    @classmethod
+    def get_header(cls, data):
+        return cls.header
+
+    @classmethod
+    def get_row(cls, i, row_data):
+        name_template = '``"{}"``'
+        names = [row_data["name"]] + row_data["synonyms"]
+        name = " or ".join(name_template.format(n) for n in names)
+        return cls.row_template.format(name, row_data["hex"], row_data["hex"])
+
+
+def make_all():
```

Comment:
```suggestion
def make_all_tables():
```
The expression `make_all` is ambiguous in scope, so how about `make_all_tables`? We also need to change `conf.py` too.